core/parameters/src/config.rs

```diff
@@ -208,3 +208,17 @@ impl CongestionControlConfig {
         }
     }
 }
+
+/// Configuration specific to ChunkStateWitness.
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub struct WitnessConfig {
+    /// Size limit for storage proof generated while executing receipts in a chunk.
+    /// After this limit is reached we defer execution of any new receipts.
+    pub main_storage_proof_size_soft_limit: usize,
+    /// Maximum size of transactions contained inside ChunkStateWitness.
+    /// A witness contains transactions from both the previous chunk and the current one.
+    /// This parameter limits the sum of sizes of transactions from both of those chunks.
+    pub combined_transactions_size_limit: usize,
```

Comment:
Yeah it could happen with a lot of large transactions, but IMO this approach should be good enough. The chunk producers are assigned at random, so it's not like someone can use this mechanism for anything malicious.